packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -265,9 +304,38 @@ export default class Extension {
     return true;
   }
 
+  private derive (parentAddress: string, suri: string, password: string, metadata: KeyringPair$Meta): KeyringPair {
+    const parentPair = keyring.getPair(parentAddress);
+    try {
+      parentPair.decodePkcs8(password);
+    } catch (e) {
+      throw new Error('invalid password');
+    }
+    try {
+      return parentPair.derive(suri, metadata);
+    } catch (err) {
+      throw new Error(`"${suri}" is not a valid derivation path`);
+    }
+  }
+
+  private derivationValidate ({ parentAddress, suri, parentPassword }: RequestDeriveValidate): ResponseDeriveValidate {
+    const childPair = this.derive(parentAddress, suri, parentPassword, {});
+    return {
+      address: childPair.address,
+      suri
+    };
+  }
+
+  private derivationCreate ({ parentAddress, suri, parentPassword, genesisHash, name, password }: RequestDeriveCreate): boolean {
+    const childPair = this.derive(parentAddress, suri, parentPassword, { genesisHash, name });
```

Comment:
here I would add the `parentAddress` to the meta as well, in addition to the `genesisHash` & `name` (meta can take anything) - this means we can actually show links between accounts, i.e. "parent is...". (Don't suggest it here as of now, but lets make it possible)